game/index.html

```diff
@@ -1,55 +1,167 @@
+            let iOS = isIOS();
+            console.log("isSafari: ", safari, "isIOS: ", iOS);
+            if (safari) {
+                setElementVisibility('unsupported-safari-ios', iOS);
+                setElementVisibility('unsupported-safari-mac', !iOS);
+            }
+        }
+
+        async function fetchWithProgress() {
+            setElementVisibility('progress', true);
+            setElementVisibility('unsupported', false);
+            const t0 = performance.now();
+            console.log("Started loading WASM");
+            let response = await fetch('./game_bg.wasm');
+            const contentLength = response.headers.get('Content-Length');
+            const reader = response.body.getReader();
+            let receivedLength = 0;
+            let chunks = [];
+            while (true) {
+                const {done, value} = await reader.read();
+                if (done) {
+                    break;
+                }
+                chunks.push(value);
+                receivedLength += value.length;
+                document.getElementById("progress_text").innerText = prettyPrintBytes(receivedLength) + " / " + prettyPrintBytes(contentLength);
+                document.getElementById("progress_bar").style.width = (100.0 * receivedLength / contentLength) + "%";
+            }
+            document.getElementById("progress_text").innerText = "Loaded " + prettyPrintBytes(contentLength) + ", now initializing WASM module";
+            let blob = new Blob(chunks);
+            let buffer = await blob.arrayBuffer();
+            const t1 = performance.now();
+            console.log(`It took ${t1 - t0} ms to download WASM, now initializing it`);
+            await init(buffer);
+        }
+
+        main();
+    </script>
+    <style type="text/css">
+        body {
+            background-color: white;
+        }
+        #loading {
+            background-color: #94C84A;
+            padding: 40px;
+            color: black;
+            font-family: arial;
+            border: solid black 3px;
+            border-radius: 4px;
+            max-width: 500px;
+            margin: auto;
+        }
+        #loading h1 {
+            text-align: center;
+        }
+        #unsupported-proceed-btn {
+            display: block;
+            margin: auto;
+        }
+    </style>
 </head>
-<body style="background-color:black;">
-	<div id="loading" style="padding-top: 40px; color: white; text-align: center; font-family: arial; font-size: 200%;">
-		<h1>Loading A/B Street...</h1>
-		<div style="width: 100%; background-color: white;">
-			<div style="width: 1%; height: 30px; background-color: red;" id="progress_bar"></div>
-		</div>
-		<div id="progress_text"></div>
-		<p>If you think something has broken, check your browser's developer console (Ctrl+Shift+I or similar)</p>
-		<p>(Your browser must support WebGL and WebAssembly)</p>
-	</div>
+<body>
+    <div id="loading" >
+        <h1>A/B<br />Street</h1>
```

Comment:
Why the
dramatic newline?

Reply:
lol because it's dramatic, I guess. 
![](https://avatars.githubusercontent.com/u/78323823?s=200&v=4)
You're right though, it looks were outside of the context of the actual logo. 

I could plop in the actual logo, but would prefer to keep the various non-abstreet loading screens similar.

Reply:
Yeah, I vote for text and no newline. Thanks for explaining the source of it -- the same effect doesn't carryover without that arrow ;)